app/models/post.rb

```diff
@@ -1,3 +1,11 @@
 class Post < ApplicationRecord
-  belongs_to :user
+  belongs_to :author, class_name: 'User'
+
+  def updates_posts_counter
+    posts.increment!(:Posts_Counter)
+  end
```

Comment:
### OPTIONAL
You can use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - after_save to call the #update_post_counter method every time a new post has been saved to the database, then make the update_post_counter method private

```ruby
after_save :update_post_counter

  private

  def update_post_counter
    author.increment!(:posts_counter)
  end
```